### PR TITLE
Fix DateTime::getLastErrors return validation

### PR DESCRIPTION
**Pull request by AH-DL**
Do you follow the guidelines?

- [X] I have tested my changes
- [X] There is no breaking change :

Simple modification following DateTime extension docs, the return value of the method is always false when no error or warnings are stored, then the variable "$errors" should not be evaluated as an array inside the conditional.

[PHP Documentation] (https://www.php.net/manual/en/datetime.getlasterrors.php)

- [X] There is no regression
- [-] I have updated the unit tests and integration tests accordingly
- [X] I follow the existing [coding style](https://docs.kanboard.org/v1/dev/coding_standards/)

### CODE REVIEW

File: app/Core/DateParser.php [hunk 1]
```diff
@@ -204,13 +204,13 @@ private function getValidDate($value, $format)
...
             $errors = DateTime::getLastErrors();
-            if ($errors['error_count'] === 0 && $errors['warning_count'] === 0) {
+            if ($errors == false) {
                 $timestamp = $date->getTimestamp();
                 return $timestamp > 0 ? $timestamp : 0;
             }
         }
 
-        return 0;
+        return false;
```
Comment (ChrisCarini) [inline app/Core/DateParser.php:213]: should this change too? if so, it might be good to update the doc string

Comment (AH-DL) [reply app/Core/DateParser.php:213]: I think so as it is a static value, docstring updated :+1:  .

Comment (ChrisCarini) [reply app/Core/DateParser.php:213]: I'm not that familiar with this code base, but my inclination is to think that int return type is correct, and not boolean - the other exit location appears to return an integer unless I am misunderstanding the method :)

Comment (AH-DL) [reply app/Core/DateParser.php]: Thanks for the atention! its enough with the conditional fix, opened new [PR](https://github.com/kanboard/kanboard/pull/5116) to address that.

**Pull request closed without merging**